src/NuGet.Clients/NuGet.CommandLine/MsBuildUtility.cs

```diff
@@ -258,7 +260,8 @@ public static bool IsMsBuildBasedProject(string projectFullPath)
             AddPropertyIfHasValue(args, "RestoreSolutionDirectory", solutionDirectory);
             AddPropertyIfHasValue(args, "RestoreConfigFile", restoreConfigFile);
             AddPropertyIfHasValue(args, "RestorePackagesPath", packagesDirectory);
-            AddPropertyIfHasValue(args, "SolutionDir", solutionDirectory);
+            AddPropertyIfHasValue(args, "SolutionDir", solutionDirectory != null ? PathUtility.EnsureTrailingSlash(solutionDirectory) : null);
```

Comment:
I don't see why this should be any different from RestoreSolutionDirectory, since they rely on the same value. 

You can do the trailing slash appending in the GetSolutionDirectory call in the restore command.